app/components/MarkersManager.tsx

```diff
@@ -2,69 +2,313 @@ import { Marker, Popup, useMapEvents } from "react-leaflet";
-                    </Marker>
-                    <Marker position={redPosition} icon={RedMarker} interactive={true} eventHandlers={{ click: () => setRedPosition(null) }}>
-                    </Marker>
-                </>
-            )
-        }
-        else if (greenPosition !== null) {
-            return (
-                <Marker position={greenPosition} icon={GreenMarker} interactive={true} eventHandlers={{ click: () => setGreenPosition(null) }}>
-                </Marker>
-            )
-        }
-        else if (redPosition !== null) {
-            return (
-                <Marker position={redPosition} icon={RedMarker} interactive={true} eventHandlers={{ click: () => setGreenPosition(null) }}>
-                </Marker>
-            )
+  setOrigin: (origin: string) => void;
+  setDestination: (destination: string) => void;
+  creatingRoute: boolean;
+}) {
+  // close modal when ESC key is pressed
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLButtonElement>) => {
+    if (e.key === "Escape") {
+      e.preventDefault();
+      // check which modal is open
+      if (isModalOpen) {
+        closeModal();
+      } else if (isPOIModalOpen) {
+        closePOIModal();
+      }
+    }
+  };
+
+  const [redPosition, setRedPosition] = useState<LatLng | null>(null);
+  const [greenPosition, setGreenPosition] = useState<LatLng | null>(null);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [isPOIModalOpen, setIsPOIModalOpen] = useState<boolean>(false);
+  const [canCreateMarker, setCanCreateMarker] = useState<boolean>(true); // set to true at the beginning
+
+  let [poiLat, setPoiLat] = useState<number>(0);
+  let [poiLon, setPoiLon] = useState<number>(0);
+
+  // close modal when btn is clicked
+  const closeModal = () => {
+    setIsModalOpen(false);
+    setCanCreateMarker(false);
+  };
+
+  const closePOIModal = () => {
+    setIsPOIModalOpen(false);
+    closeModal();
+  };
+  const storePos = (lat: number, lon: number) => {
+    setPoiLat(lat);
+    setPoiLon(lon);
+  };
+
+  const createPOI = () => {
+    console.log("create POI");
+    console.log("lat:", poiLat.toFixed(3));
+    console.log("lon:", poiLon.toFixed(3));
+    // open modal to create POI
+    setIsPOIModalOpen(true);
+    closeModal();
+  };
+
+  const savePOI = () => {
+    console.log("save POI to DB");
+    // after saving POI, close modals
+    // get name and type from form
+    const name = (document.getElementById("name") as HTMLInputElement).value;
+    const type = (document.getElementById("type") as HTMLInputElement).value;
+
+    // validate name and type are not empty strings
+    if (name === "" || type === "") {
+      closePOIModal();
+      return;
+    }
+
+    // validate that name and type are not too long
+    else if (name.length > 20 || type.length > 20) {
+      closePOIModal();
+      return;
+    }
```

Comment:
Why not display a message to the user instead of closing the model?

Reply:
had an alert (testing only), mas dps apaguei.
Will implement a form control, ty for the reminder :)